### ISSUE TITLE
requests.exceptions.RetryError

### ISSUE THREAD

**Issue by Feng-Zhang** [EGA-archive/ega-download-client#173]
when I run `pyega3 -cf ../raw-data/CREDENTIALS_FILE.json fetch EGAD00001006645`, the error was throwed: 
```
[2022-09-28 17:16:06 +0800] 
[2022-09-28 17:16:06 +0800] pyEGA3 - EGA python client version 4.0.5 (https://github.com/EGA-archive/ega-download-client)
[2022-09-28 17:16:06 +0800] Parts of this software are derived from pyEGA (https://github.com/blachlylab/pyega) by James Blachly
[2022-09-28 17:16:06 +0800] Python version : 3.10.5
[2022-09-28 17:16:06 +0800] OS version : Linux #138-Ubuntu SMP Wed Jun 22 15:00:31 UTC 2022
[2022-09-28 17:16:06 +0800] Server URL: https://ega.ebi.ac.uk:8052/elixir/data
[2022-09-28 17:16:06 +0800] Session-Id: 3795056343
[2022-09-28 17:16:10 +0800] 
[2022-09-28 17:16:10 +0800] Authentication success for user 'XX@XX'
Traceback (most recent call last):
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/requests/adapters.py", line 489, in send
    resp = conn.urlopen(
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/urllib3/connectionpool.py", line 878, in urlopen
    return self.urlopen(
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/urllib3/connectionpool.py", line 878, in urlopen
    return self.urlopen(
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/urllib3/connectionpool.py", line 878, in urlopen
    return self.urlopen(
  [Previous line repeated 7 more times]
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/urllib3/connectionpool.py", line 868, in urlopen
    retries = retries.increment(method, url, response=response, _pool=self)
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='ega.ebi.ac.uk', port=8052): Max retries exceeded with url: /elixir/data/metadata/datasets/EGAD00001006645/files (Caused by ResponseError('too many 500 error responses'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/bin/pyega3", line 10, in <module>
    sys.exit(main())
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/pyega3/pyega3.py", line 148, in main
    execute_subcommand(args, data_client)
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/pyega3/libs/commands.py", line 19, in execute_subcommand
    list_files_in_dataset(args, data_client)
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/pyega3/libs/commands.py", line 68, in list_files_in_dataset
    files = dataset.list_files()
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/pyega3/libs/data_set.py", line 49, in list_files
    reply = self.data_client.get_json(f"/metadata/datasets/{self.id}/files")
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/pyega3/libs/data_client.py", line 55, in get_json
    r = self.session.get(url, headers=headers)
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
  File "/home/zhangfeng/bin/miniconda3/envs/bio-py/lib/python3.10/site-packages/requests/adapters.py", line 556, in send
    raise RetryError(e, request=request)
requests.exceptions.RetryError: HTTPSConnectionPool(host='ega.ebi.ac.uk', port=8052): Max retries exceeded with url: /elixir/data/metadata/datasets/EGAD00001006645/files (Caused by ResponseError('too many 500 error responses'))

```

I have no idea how to fix this. Is it a port problem? I am not sure whether it is open:
```
Status: active
Logging: on (low)
Default: deny (incoming), allow (outgoing), deny (routed)
New profiles: skip

To                         Action      From
--                         ------      ----
8443                       ALLOW IN    Anywhere                  
8052                       ALLOW IN    Anywhere                  
8443/tcp                   ALLOW IN    Anywhere                  
8052/tcp                   ALLOW IN    Anywhere                  
8443 (v6)                  ALLOW IN    Anywhere (v6)             
8052 (v6)                  ALLOW IN    Anywhere (v6)             
8443/tcp (v6)              ALLOW IN    Anywhere (v6)             
8052/tcp (v6)              ALLOW IN    Anywhere (v6)
```

**Comment by CsabaHalmagyi** [comment]
Hi @Feng-Zhang 
Thank you for reporting this. We are currently experiencing some issues with the underlying storage service, the python client's 500 error is a reflection of that. The technical team is working on a solution.

**Comment by regmdr** [comment]
Hello,
I have been having the same issue for several days now and I was wondering if this has been resolved from your end?
Thank you for the updates!

**Comment by Feng-Zhang** [comment]
Not yet. But I run this command again, it works sometimes. What only I can do is repeat the command after encountering an error.

**Comment by AnnaK135** [comment]
Hello!
Are there any possibilities to overcome this issue or download the data otherwise? I've been trying for a couple of weeks already and the issue still does not seem to be resolved. 
(I have used the EGA client before with the same credentials and environment and it worked well)

**Comment by f6v** [comment]
I'm seeing this error as well. Now the banner at EGA landing page is gone and I assumed I could download the dataset. Alas, it's been hours and I can't get even one 6 GB fasta file. 

I understand that something is really really wrong at EGA and it might be beyond the control of people maintaining this client. But I think it'd be fair to be more transparent. It's crystal clear that the issue(s) critical to the service stability is still there, but there's no heads-up anywhere. The support seems to be stretched thin as well.

**Comment by malloryfreeberg** [comment]
Thank you, all, for your patience. The underlying storage system issues have been resolved, and we have also been working on and released a new version of both [pyega3 (v5.0.0)](https://pypi.org/project/pyega3/) and the underlying API. Please update to the newest version of pyega3 (you can find the instructions [here](https://github.com/EGA-archive/ega-download-client#installation-and-update)) to enjoy the new features, including a more stable service which results in a higher download success rate.

This GH issue will be closed, but if you continue to experience the same problem, feel free to re-open it. If you experience any other problems, please email the EGA Helpdesk at helpdesk@ega-archive.org and attach the output log produced by pyega3 (pyega3_output.log). The Helpdesk team can then provide further assistance. Thank you!
